app/models/sms_message.rb

```diff
@@ -52,4 +52,14 @@ class SmsMessage < ApplicationRecord
                   received: "received",
                   read: "read",
   }, prefix: :status
+
+  def status=(new_status)
+    new_index = SmsMessage.statuses.keys.to_a.index(new_status)
```

Comment:
This is gross but I didn't want to re-declare the statuses in a separate list, and the keys method seems to keep the order from the dictionary-like way we declare them above here. I'm fine with changing it to a static array if you'd prefer.